### PR TITLE
[SYCL][NFC] Clean up options in aot/multiple-devices.cpp test commands

### PR DESCRIPTION
**Pull request by AGindinson**
Any specific AOT target triples prove obsolete for the commands that
are used to produce SPIR-V, the latter being completely device-agnostic.
Subarch-specific AOT compilers will accept it just as well.

Signed-off-by: Artem Gindinson <artem.gindinson@intel.com>

### CODE REVIEW

**Comment by bader** [comment]
> Any specific AOT target triples prove obsolete for the commands that
> are used to produce SPIR-V, the latter being completely device-agnostic.
> Subarch-specific AOT compilers will accept it just as well.
> 

I'm not sure I follow this. 
According to my understanding current command line options are supposed to enable ahead-of-time compilation for x86, GEN and FPGA targets. Am I wrong?

**Comment by AGindinson** [comment]
> According to my understanding current command line options are supposed to enable ahead-of-time compilation for x86, GEN and FPGA targets. Am I wrong?

If the test used a 1-command compilation, this would've been entirely correct.
```
clang++ -fsycl-targets=spir64_x86_64-*,spir64_gen-*,spir64_fpga-* app.cpp
```
This would've told the driver: "Please, produce object file, SPIR-V, and then process the spirv with the corresponding AOT tools: for x86, for GEN, for FPGA. Then, link the resulting device images into the final fat binary"

However, in this particular test all stages are run manually. In particular, the AOT tools are called explicitly (lines 14-16), and the SPIR-V fed into those is explicitly produced, too. Then, the explicit calls to the linker pass the resulting device images through the `-fsycl-add-targets` option, and that's where the triples are specified with the aim of storing the image target information for image selection at runtime.

To sum up, mentioning specific subarch triple(s):
a) makes driver call the device-specific AOT compiler(s) after producing SPIR-V (called manually in this test);
b) passes image target information to clang-offload-wrapper, which will in turn pass it to the fat binary so that runtime selection can be performed (done through `-fsycl-add-targets`).

**Comment by bader** [comment]
Do we have a test using '1-command compilation'?
If not, I would "clean-up" the test by replacing "manual commands" (which should be covered by the clang lit tests) by '1-command compilation'.

**Comment by AGindinson** [comment]
> Do we have a test using '1-command compilation'?
I've edited my initial comment - one-device AOT tests are doing just that.

> If not, I would "clean-up" the test by replacing "manual commands" (which should be covered by the clang lit tests) by '1-command compilation'.
The idea here is to get all types of device-specific images combinations so that the runtime selection mechanism can be thoroughly tested. This is why I prefer to produce the device images explicitly, and then throw in all kinds of linker commands to receive different combinations without having to re-compile the source each time.

I can add a separate 1-line run for all devices if you think we should have that as well.

**Comment by bader** [comment]
> I can add a separate 1-line run for all devices if you think we should have that as well.

I think this is what we recommend for the compiler users. Right?
If so, we must test this way.
I think current way of testing this functionality is optional and don't have much value for end users unless they are doing something esoteric.

File: sycl/test/aot/multiple-devices.cpp [hunk 1]
```diff
@@ -8,9 +8,18 @@
 
 // REQUIRES: ioc64, ocloc, aoc
 
-// Produce object file, spirv
-// RUN: %clangxx -fsycl -fsycl-targets=spir64_x86_64-unknown-linux-sycldevice,spir64_gen-unknown-linux-sycldevice,spir64_fpga-unknown-linux-sycldevice %s -c -o %t.o
-// RUN: %clangxx -fsycl -fsycl-link-targets=spir64_x86_64-unknown-linux-sycldevice,spir64_gen-unknown-linux-sycldevice,spir64_fpga-unknown-linux-sycldevice %t.o -o %t.spv
+// 1-command compilation case
+// Targeting CPU, GPU, FPGA
+// RUN: %clangxx -fsycl -fsycl-targets=spir64_x86_64-unknown-linux-sycldevice,spir64_gen-unknown-linux-sycldevice,spir64_fpga-unknown-linux-sycldevice -Xsycl-target-backend=spir64_gen-unknown-linux-sycldevice "-device skl" %s -o %t_all.out
```
Comment (bader) [inline sycl/test/aot/multiple-devices.cpp]: Just note: "-device skl" means this test won't work on older SKUs. Right?

Comment (AGindinson) [reply sycl/test/aot/multiple-devices.cpp]: I'm not quite sure, TBH - I've never tried running this on older architectures.
Depending on how our AOT tools change/our compiler options expand in foreseeable future, we may get better ways of specifying the GEN arch.

Comment (bader) [reply sycl/test/aot/multiple-devices.cpp]: Most likely GEN compiler will utilize Gen9 ISA. Can we use `broadwell` instead?
